.circleci/config.yml

```diff
@@ -1,85 +1,193 @@
+                       -m "Branch: ${CIRCLE_BRANCH}, commit: ${CIRCLE_SHA1}, tag:${CIRCLE_TAG-none}"
+
+            # push changes to the Github Pages publishing branch on origin repos
+            git push --set-upstream origin $BRANCH
+            git push origin
+
+jobs:
+  styleCheck:
+    executor: node_executor
+    steps:
+      - prepare_node
       - run:
-          name: Unit tests
-          command: npm run test:coverage
+          name: Check lints
+          command: npm run lint
+      - run:
+          name: Check formatting
+          command: npm run check-formatting
+
+  build:
+    executor: node_executor
+    steps:
+      - prepare_node
       - run:
           name: Build
           command: npm run build
-  publish:
-    docker:
-      - image: circleci/node:12
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - ./lib
+
+  test:
+    executor: node_executor
+    steps:
+      - prepare_node
+      - run:
+          name: Unit tests
+          command: npm run test:coverage
+
+  publishSdk:
+    executor: node_executor
     steps:
       - checkout
       - restore_cache:
           keys:
-            - v1-dependencies-{{ checksum "package.json" }}
-            # fallback to using the latest cache if no exact match is found
-            - v1-dependencies-
-      - run:
-          name: Build
-          command: npm run build
-      - run:
-          name: Publish on npm
+            - node-cache-{{ checksum "package-lock.json" }}
+            - node-cache-
+      - attach_workspace:
+          at: ~/project
+      - deploy:
+          name: Publish sdk package on NPM
           command: |
             echo '//registry.npmjs.org/:_authToken=${NPM_TOKEN}'>.npmrc
             npm publish --access public
 
+  generateDoc:
+    executor: node_executor
+    steps:
+    - prepare_node
+    - run:
+        name: build doc
+        command: npm run generate:docs
+    - run:
+        name: archive doc
+        command: tar -czf orchestrate-sdk-doc.tgz ./docs/out
+    - store_artifacts:
+        path: ./orchestrate-sdk-doc.tgz
+    - run:
+        name: organise doc dirs
+        command: |
+          node .circleci/organise_doc.js --rootdir=$(pwd) --tag="${CIRCLE_TAG}"
+          ls -la docs/
+    - persist_to_workspace:
+        root: ~/project
+        paths:
+          - ./docs
+
+  publishDocMaster:
+    executor: node_executor
+    steps:
+    - publish_doc
+
+  publishDocTag:
+    executor: node_executor
+    steps:
+    - publish_doc
+
 workflows:
   version: 2
   default:
     jobs:
-      - build: # Runs for the specified branches OR all tags
+      - styleCheck
+      - build
+      - test
+      - generateDoc:
+          requires:
+            - build
+      - publishDocMaster:
+          requires:
+            - styleCheck
+            - test
+            - generateDoc
           filters:
             branches:
-              only:
```

Comment:
Seems that this was removed? The regex on branch names. Maybe not needed

Reply:
Yes it looks strange in this diff but the change is the following:
- build job now run all the time without restrictions, it's a fast job and very useful to smoke test. Also it's required in the CI steps in Github so if it doesn't run on PR it blocks the PR.
- PublishDocMaster only happens on master. Doc should not be published when building a PR, but only once merged.

Reply:
Perfect